proctor-common/src/main/java/com/indeed/proctor/common/model/Allocation.java

```diff
@@ -37,13 +40,24 @@ public Allocation(@Nullable final String rule,
         this.id = Strings.nullToEmpty(id);
```

Comment:
should it calls the new constructor with id and name?

Reply:
Yes, it should call the new constructor with the id and name. Still keep some previous constructors, so if the client doesn't send the name field, it still works as well

Reply:
sounds good.

Just to clarify, i was asking whether this method should become this(rules, changes, id, Strings.EMPTY)?

Reply:
Good point. I think so. It should become this (rules, range, id, "")
I'll make the change

Reply:
I updated